src/include/add-on/add-on-workflow.rb

```diff
@@ -224,37 +217,31 @@ def MediaSelect
         sources_after = Pkg.SourceGetCurrent(false)
         Builtins.y2milestone("Sources with new one added: %1", sources_after)
 
-        # bnc #393011
-        # AddOnProduct::src_id must be set to the latest source ID
-        src_id_found = false
+        @added_repos = []
 
-        Builtins.foreach(sources_after) do |one_source|
-          if !Builtins.contains(sources_before, one_source)
-            AddOnProduct.src_id = one_source
-            Builtins.y2milestone("Added source ID is: %1", AddOnProduct.src_id)
-            src_id_found = true
-            raise Break
-          end
+        sources_after.each do |source|
+          next if sources_before.include?(source)
+
+          @added_repos << source
+          log.info("Added source ID: #{source}")
         end
 
-        if src_id_found
-          # BNC #481828: Using LABEL from content file as a repository name
-          Packages.AdjustSourcePropertiesAccordingToProduct(AddOnProduct.src_id)
-          # used add-ons are stored in a special list
-          AddAddOnToStore(AddOnProduct.src_id)
+        if !@added_repos.empty?
+          @added_repos.each do |src_id|
+            # BNC #481828: Using LABEL from content file as a repository name
+            Packages.AdjustSourcePropertiesAccordingToProduct(src_id)
+            # used add-ons are stored in a special list
+            AddAddOnToStore(src_id)
+          end
         else
-          AddOnProduct.src_id = Ops.get(
-            sources_after,
-            Ops.subtract(Builtins.size(sources_after), 1),
-            0
-          )
-          Builtins.y2warning("Fallback src_id: %1", AddOnProduct.src_id)
+          @added_repos << sources_after.last
+          log.warn("Fallback src_id: #{sources_after.last}")
```

Comment:
I have no idea what is going here. Can you please add comment explaining this?

Reply:
Hmm, no idea, IMHO this should never happen. I just added a FIXME note, I do not want to break something...